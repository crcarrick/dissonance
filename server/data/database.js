import { Model, Sequelize } from 'sequelize';

import { channel } from './channel/channel.model';
import { message } from './message/message.model';
import { server } from './server/server.model';
import { user } from './user/user.model';

export const connectDatabase = () => {
  const sequelize = new Sequelize(
    process.env.DATABASE_NAME,
    process.env.DATABASE_USERNAME,
    process.env.DATABASE_PASSWORD,
    {
      host: 'localhost',
      dialect: 'postgres',
      logging: false,
    }
  );

  const Channel = channel({ sequelize });
  const Message = message({ sequelize });
  const Server = server({ sequelize });
  const User = user({ sequelize });

  class UserServer extends Model {}

  UserServer.init({}, { sequelize, tableName: 'users_servers' });

  const models = { Channel, Message, Server, User, UserServer };

  Object.keys(models).forEach((key) => {
    if ('associate' in models[key]) {
      models[key].associate(models);
    }
  });

  return { models, sequelize };
};
