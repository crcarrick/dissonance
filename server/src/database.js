import { merge } from 'lodash';

import knexfile from '../knexfile';

export const createDbClient = (knex, database) =>
  database
    ? knex(merge(knexfile[process.env.NODE_ENV], { connection: { database } }))
    : knex(knexfile[process.env.NODE_ENV]);
