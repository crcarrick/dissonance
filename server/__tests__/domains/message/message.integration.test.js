import { gql, PubSub } from 'apollo-server';

import { createDbClient } from '@dissonance/database';

import {
  createTestClient,
  generateMockData,
  seedDatabase,
} from '@dissonance/test-utils';

const knex = jest.requireActual('knex');

describe('Message Integration', () => {
  const { users, channels } = generateMockData();

  let dbClient;
  beforeAll(() => {
    dbClient = createDbClient(
      knex,
      `dissonance_test_${process.env.JEST_WORKER_ID}`
    );
  });

  afterAll(async () => {
    await dbClient.destroy();
  });

  let gqlClient;
  beforeEach(async () => {
    await seedDatabase(dbClient);

    gqlClient = createTestClient({
      context: {
        user: users[0],
        pubsub: new PubSub(),
      },
      dbClient,
    });
  });

  test('creates a message', async () => {
    const response = await gqlClient.mutate({
      mutation: gql`
        mutation CreateMessage($input: CreateMessageInput!) {
          createMessage(input: $input) {
            id
            text
          }
        }
      `,
      variables: {
        input: {
          channelId: channels[0].id,
          text: 'Testing 123',
        },
      },
    });

    expect(response.data.createMessage).toMatchSnapshot({
      id: expect.any(String),
    });
  });
});
