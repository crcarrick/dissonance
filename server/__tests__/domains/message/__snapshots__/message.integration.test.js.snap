// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Integration creates a message 1`] = `
Object {
  "id": Any<String>,
  "text": "Testing 123",
}
`;
